src/__tests__/requirement07.test.js

```diff
@@ -1,24 +1,19 @@
 import React from 'react';
 import { render, screen, waitFor, fireEvent } from '@testing-library/react';
 import App from '../App';
-import * as api from '../services/api';
-import mockedCategoriesResult from '../__mocks__/categories';
 import mockedQueryResult from '../__mocks__/query';
-
-jest.mock('../services/api');
-api.getCategories.mockImplementation(
-  () => Promise.resolve(mockedCategoriesResult),
-);
-api.getProductsFromCategoryAndQuery.mockImplementation(
-  () => Promise.resolve(mockedQueryResult),
-);
+import mockFetch from '../__mocks__/mockFetch';
 
 describe(`7 - Redirecione para uma tela com a exibição detalhada ao clicar na exibição resumida de um produto`, () => {
   it('Clicar no card produto leva à página com seus detalhes', async () => {
+
+    jest.spyOn(global, 'fetch').mockImplementation((url) => {
+      return mockFetch(url);
+    });
     render(<App />);
-    await waitFor(() => expect(api.getCategories).toHaveBeenCalled());
+    await waitFor(() => expect(global.fetch).toHaveBeenCalled());
     fireEvent.click(screen.getAllByTestId('category')[0]);
-    await waitFor(() => expect(api.getProductsFromCategoryAndQuery).toHaveBeenCalled());
+    await waitFor(() => expect(global.fetch).toHaveBeenCalled());
```

Comment:
Mesma coisa aqui  :) 

```suggestion
    await waitFor(() => expect(global.fetch).toHaveBeenCalledTimes(2));
```